serialization-tests/src/test/java/org/eclipse/collections/impl/map/ordered/mutable/UnmodifiableMutableOrderedMapSerializationTest.java

```diff
@@ -21,16 +21,31 @@
     @Test
     public void serializedForm()
     {
-        Verify.assertSerializedForm(
-                1L,
-                "rO0ABXNyAE5vcmcuZWNsaXBzZS5jb2xsZWN0aW9ucy5pbXBsLm1hcC5vcmRlcmVkLm11dGFibGUu\n"
-                        + "VW5tb2RpZmlhYmxlTXV0YWJsZU9yZGVyZWRNYXAAAAAAAAAAAQIAAUwACGRlbGVnYXRldAAzTG9y\n"
-                        + "Zy9lY2xpcHNlL2NvbGxlY3Rpb25zL2FwaS9tYXAvTXV0YWJsZU9yZGVyZWRNYXA7eHBzcgBCb3Jn\n"
-                        + "LmVjbGlwc2UuY29sbGVjdGlvbnMuaW1wbC5tYXAub3JkZXJlZC5tdXRhYmxlLk9yZGVyZWRNYXBB\n"
-                        + "ZGFwdGVyAAAAAAAAAAECAAFMAAhkZWxlZ2F0ZXQAD0xqYXZhL3V0aWwvTWFwO3hwc3IAF2phdmEu\n"
-                        + "dXRpbC5MaW5rZWRIYXNoTWFwNMBOXBBswPsCAAFaAAthY2Nlc3NPcmRlcnhyABFqYXZhLnV0aWwu\n"
-                        + "SGFzaE1hcAUH2sHDFmDRAwACRgAKbG9hZEZhY3RvckkACXRocmVzaG9sZHhwP0AAAAAAAAB3CAAA\n"
-                        + "ABAAAAAAeAA=",
-                UnmodifiableMutableOrderedMap.of(OrderedMaps.adapt(new LinkedHashMap<>())));
+        if (this.isBeforeJava21())
+        {
+            Verify.assertSerializedForm(
+                    1L,
+                    "rO0ABXNyAE5vcmcuZWNsaXBzZS5jb2xsZWN0aW9ucy5pbXBsLm1hcC5vcmRlcmVkLm11dGFibGUu\n"
+                            + "VW5tb2RpZmlhYmxlTXV0YWJsZU9yZGVyZWRNYXAAAAAAAAAAAQIAAUwACGRlbGVnYXRldAAzTG9y\n"
+                            + "Zy9lY2xpcHNlL2NvbGxlY3Rpb25zL2FwaS9tYXAvTXV0YWJsZU9yZGVyZWRNYXA7eHBzcgBCb3Jn\n"
+                            + "LmVjbGlwc2UuY29sbGVjdGlvbnMuaW1wbC5tYXAub3JkZXJlZC5tdXRhYmxlLk9yZGVyZWRNYXBB\n"
+                            + "ZGFwdGVyAAAAAAAAAAECAAFMAAhkZWxlZ2F0ZXQAD0xqYXZhL3V0aWwvTWFwO3hwc3IAF2phdmEu\n"
+                            + "dXRpbC5MaW5rZWRIYXNoTWFwNMBOXBBswPsCAAFaAAthY2Nlc3NPcmRlcnhyABFqYXZhLnV0aWwu\n"
+                            + "SGFzaE1hcAUH2sHDFmDRAwACRgAKbG9hZEZhY3RvckkACXRocmVzaG9sZHhwP0AAAAAAAAB3CAAA\n"
+                            + "ABAAAAAAeAA=",
+                    UnmodifiableMutableOrderedMap.of(OrderedMaps.adapt(new LinkedHashMap<>())));
+        }
+    }
+
+    private boolean isBeforeJava21()
+    {
+        String javaVersion = System.getProperty("java.version").replace("-ea", "");
+        String[] javaVersionParts = javaVersion.split("\\.");
+        if ("1".equals(javaVersionParts[0]))
```

Comment:
the if is superfluous. the same will happen with the parseInt code.

Reply:
Hi @Bananeweizen , yes, the if is superfluous here, and I was debating on changing it. Something is bothering me about changing it, even though it would work.

Reply:
Ok, I've removed the if statement in the latest push. Thanks for the feedback @Bananeweizen.